kubernetes/config-maps/nginx-storage/default.conf.j2

```diff
@@ -8,12 +10,31 @@ server {
     client_max_body_size     {{ max_upload_size }};
 
 
+    set $gatekeeper_api_backend http://gatekeeper.default.svc.cluster.local;
+
     location /download/ {
-        limit_except GET { deny all; }
+        limit_except        GET { deny all; }
+        auth_request        /download-auth/;
+        auth_request_set    $auth_status $upstream_status;
+        if ($request_uri ~* "/download/(.*)") {
+            set $download_url $1;
+        }
+        error_page 401 = @error_401;
         alias /srv/storage/;
     }
 
+    location /download-auth/ {
+        internal;
+        proxy_pass                $gatekeeper_api_backend/gatekeeper/download-auth/$download_url;
+        proxy_pass_request_body   off;
+        proxy_set_header          Content-Length "";
```

Comment:
According to nginx site: https://www.nginx.com/resources/admin-guide/restricting-access-auth-request/ request body is discarded for authentication subrequests, so we will need to removing the body and `Content-Length`.

Reply:
But if it's discarded, it should not be necessary, right? What happens in gatekeeper if you remove these lines?

Reply:
I think it's work, but I not sure what happend, because gatekeeper not work properly now(problem with golem token). I delete it.

Reply:
I not sure what happend, because gatekeeper not work properly now(problem with golem token)
Without this header, I occur additional errors:
- download-auth: `[DEBUG] Ignoring connection reset`
- update-auth: `ERROR | Request body exceeded settings.DATA_UPLOAD_MAX_MEMORY_SIZE`

In update-auth looks like subrequest send file in body to the django.
Smaller file occur next problem:
- `[DEBUG] Ignoring EPIPE`
I think it not work porperly without this header.

Reply:
OK. Looks like `auth_request` does not remove the body after all. So we do have to remove it.